lib/ssl/src/tls_connection.erl

```diff
@@ -741,24 +745,27 @@ hello(internal, #server_hello{extensions = Extensions} = Hello,
                  renegotiation = {Renegotiation, _},
                  ocsp_stapling_state = OcspState} = HsEnv,
 	     ssl_options = SslOptions} = State) ->
-    %% check if server has sent a certifucate status message in hello
-    IsCertStatusRequested = maps:is_key(ocsp_stapling, Extensions),
+    %% check if server has sent an empty certifucate status message in hello
+    IsCertStatusRequested = (undefined == maps:get(status_request, Extensions, false)),
```

Comment:
Same comments here, we can reuse the new function to handle all cases.

Reply:
done